kubectl-fdb/cmd/plugin_version_checker.go

```diff
@@ -0,0 +1,126 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/go-retryablehttp"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// KubectlFbReleaseURL is the public GirHub url we read the latest version
+const KubectlFbReleaseURL = "https://api.github.com/repos/FoundationDB/fdb-kubernetes-operator/releases/latest"
+
+// LocalTempVersionFileName is where we cache plugin version for 24 hours, so we don't call GitHub for every single command(also rate-limit issue on GitHub api calls)
+const LocalTempVersionFileName = "latest"
+
+// PluginVersionDetails Contains [partial] plugin version details from GitHub
+type PluginVersionDetails struct {
+	ID      int64  `json:"id"`
+	Version string `json:"tag_name"`
+	Name    string `json:"name"`
+}
+
+// It reads the latest fdb plugin version from local temp file,
+// if not exists, gets it from GitHub and store it locally and then returns it
+func getLatestPluginVersion() string {
+	fileName := filepath.Join(os.TempDir(), LocalTempVersionFileName)
+	_, err := os.Stat(fileName)
+	if os.IsNotExist(err) {
+		return updateLocalVersion(fileName)
+	}
+	versionFileIsRecent := isVersionFileCreatedToday(fileName)
+	if versionFileIsRecent {
+		return readVersionFromLocalFile(fileName)
+	}
+	_ = os.Remove(fileName)
+	return updateLocalVersion(fileName)
+}
+
+// read version value from local temp file
+func readVersionFromLocalFile(fileName string) string {
+	file, _ := os.OpenFile(fileName, os.O_RDONLY, 0644)
+	defer file.Close()
+	content, _ := os.ReadFile(fileName)
+	version := string(content)
+	return version
+}
+
+// Check if the local temp version file is created within the last 24 hours
+func isVersionFileCreatedToday(filename string) bool {
+	fileInfo, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	} else if err != nil {
+		return false
+	}
+	now := time.Now().UnixNano()
+	creationTime := fileInfo.ModTime().UnixNano()
+	diff := now - creationTime
+	hours := diff / int64(time.Hour)
+	return hours <= 24
```

Comment:
```suggestion
	return time.Since(fileInfo.ModTime()) <= 24 * time.hour
```

This should work too and would be a bit easier to understand.